File: app/actors/PriceMonitor.java
package actors;

import java.util.Map;

import com.amazonaws.services.ec2.model.InstanceType;
import com.amazonaws.services.ec2.model.SpotPrice;

public interface PriceMonitor extends AmazonClientActor
{
    public Map<InstanceType, SpotPrice> getPrices();
    
    public void monitorSpotPrices();
}

File: app/actors/PriceMonitor.scala
package actors

import scala.collection.Map
import com.amazonaws.services.ec2.model.InstanceType
import com.amazonaws.services.ec2.model.SpotPrice
import model.SpotPriceInfo

trait PriceMonitor extends AmazonClientActor {

	def getPrices(): Map[InstanceType, SpotPriceInfo];
    
    def monitorSpotPrices(): Unit;
}